src/main/java/wis/Ui.java

```diff
@@ -0,0 +1,144 @@
+package wis;
+
+import java.time.format.DateTimeParseException;
+import java.util.InputMismatchException;
+
+import wis.task.Task;
+import wis.task.Todo;
+import wis.task.Deadline;
+import wis.task.Event;
+import wis.task.TaskList;
+import wis.util.InputParser;
+import wis.util.WisException;
+import wis.util.Printer;
+
+public class Ui {
+    private String input;
+
+    public Ui(String input) {
+        this.input = input;
+    }
+
+    public void setInput(String input) {
+        this.input = input;
+    }
+
+    public boolean parseInput(TaskList tasks) {
+        String[] words = input.split(" ");
+        Action action = InputParser.parseAction(input, words);
+        switch (action) {
+        case NONE:
+            break;
+        case ADD_TODO:
+            addTodo(tasks);
+            break;
+        case ADD_DEADLINE:
+            addDeadline(tasks);
+            break;
+        case ADD_EVENT:
+            addEvent(tasks);
+            break;
+        case MARK:
+            mark(words, tasks);
+            break;
+        case UNMARK:
+            unmark(words, tasks);
+            break;
+        case DELETE:
+            delete(words, tasks);
+            break;
+        case LIST:
+            Printer.printActionAttach(Action.LIST, tasks);
+            break;
+        case BYE:
+            return true;
+        case INVALID:
+            WisException.ActionExceptionHandler(Action.INVALID);
+            break;
+        default:
+            throw new IllegalArgumentException("Illegal action argument provided.\n");
+        }
+        return false;
+    }
+
+    private void addTodo(TaskList tasks) {
+        try {
+            Todo todo = new Todo(InputParser.parseTodo(input));
+            tasks.add(todo);
+            Printer.printActionAttach(Action.ADD_TODO, todo, tasks);
+        } catch (InputMismatchException e) {
+            WisException.ActionExceptionHandler(Action.ADD_TODO);
+        }
+    }
+
+    private void addDeadline(TaskList tasks) {
+        try {
+            String[] parsedString = InputParser.parseDeadline(input);
+            Deadline deadline = new Deadline(parsedString[0],
+                    InputParser.parseDateTime(parsedString[1]));
+            tasks.add(deadline);
+            Printer.printActionAttach(Action.ADD_DEADLINE, deadline, tasks);
+        } catch (InputMismatchException e) {
+            WisException.ActionExceptionHandler(Action.ADD_DEADLINE);
+        } catch (DateTimeParseException e) {
+            WisException.DateTimeExceptionHandler(e);
+        } catch (ArrayIndexOutOfBoundsException e) {
+            WisException.DateTimeExceptionHandler(e);
```

Comment:
Could consider catching both exceptions in the same catch block since handling of the exception is similar

```java
        } catch (DateTimeParseException | ArrayIndexOutOfBoundsException  e) {
            WisException.DateTimeExceptionHandler(e);
        }
```

Reply:
This is the syntax I am looking for! I didn't know such syntax was allowed in Java.